runtime/vm/jvminit.c

```diff
@@ -6349,6 +6349,40 @@ protectedInitializeJavaVM(J9PortLibrary* portLibrary, void * userData)
 	Xj9BreakPoint("jvminit");
 #endif
 
+        {
+		IDATA enabled = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXPRINTFLAGSFINALENABLE, NULL);
+		IDATA disabled = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXPRINTFLAGSFINALDISABLE, NULL);
+		if (enabled > disabled) {
+			UDATA maxHeapSize, maxDirectMemorySize;
+			char *howset;
+
+			/*
+			 * Emulate Hotspot -XX:+PrintFlagsFinal output for two specific flags:
+			 *
+			 * [Global flags]
+			 *    size_t MaxHeapSize                              = 4294967296                                {product} {ergonomic}
+			 *  uint64_t MaxDirectMemorySize                      = 3758096384                                {product} {default}
+			 */
+
+			j9tty_printf(PORTLIB, "[Global flags]\n");
+
+			maxHeapSize = (UDATA) vm->memoryManagerFunctions->j9gc_get_maximum_heap_size(vm);
+			howset="ergonomic";
+			if ((findArgInVMArgs( PORTLIB, vm->vmArgsArray, STARTSWITH_MATCH, VMOPT_XMX, NULL, 0) >= 0) ||
+				(findArgInVMArgs( PORTLIB, vm->vmArgsArray, STARTSWITH_MATCH, MAPOPT_XXMAXHEAPSIZE_EQUALS, NULL, 0) >= 0)) {
```

Comment:
Given the mapping between `-Xmx` and `-XX:MaxHeapSize=` [1], I think we only need to search for `-Xmx` here as the `findArgInVMArgs` code handles the mapping for us.

[1] https://github.com/eclipse/openj9/blob/3774db595f1ab03626d80a7b582fe1fc13dd11d0/runtime/vm/jvminit.c#L4256